marshmallow/schema.py

```diff
@@ -668,6 +686,11 @@ def _update_fields(self, obj=None, many=False):
         if excludes:
             field_names = field_names - excludes
         ret = self.__filter_fields(field_names, obj, many=many)
+        # Set nested only parameters overriding the only parameters declared on the fields
+        for field_name, field in [(fn, f) for fn, f in ret.items() if hasattr(f, 'only')]:
+            field.only_explicit = None
+            if only_nested and field_name in only_nested:
```

Comment:
Minor: I think this can be simplified to `if field_name in only_nested:` , since `only_nested` will always be a `dict`.